### PR TITLE
Add instant mix functionality to album screen

### PR DESCRIPTION
**Pull request by 1hitsong**
**Changes**
Add Instant Mix button to album page. Clicking button will generate an instant mix based on the album and begin playing it for the user.

**Issues**
None. New functionality

### CODE REVIEW

File: source/Main.brs [hunk 1]
```diff
@@ -201,6 +201,13 @@ sub Main (args as dynamic) as void
             m.spinner = screenContent.findNode("spinner")
             m.spinner.visible = true
             group = CreateAudioPlayerGroup(screenContent.albumData.items)
+        else if isNodeEvent(msg, "instantMixSelected")
+            ' User has selected instant mix
+            ' User has selected playlist of of audio they want us to play
+            screenContent = msg.getRoSGNode()
+            m.spinner = screenContent.findNode("spinner")
+            m.spinner.visible = true
```
Comment (candry7731) [inline source/Main.brs:209]: Love the spinner!

File: source/api/Items.brs [hunk 2]
```diff
@@ -209,6 +209,17 @@ function AudioItem(id as string)
     return getJson(resp)
 end function
 
+' Get Instant Mix based on item
+function CreateInstantMix(id as string)
+    url = Substitute("/Items/{0}/InstantMix", id)
+    resp = APIRequest(url, {
+        "UserId": get_setting("active_user"),
+        "Limit": 201
```
Comment (candry7731) [inline source/api/Items.brs:217]: Why 201? It takes a really long time for my server in the cloud to compile this instant mix. Maybe my network paths are congested tonight.

**Review by candry7731 — APPROVED** [review]
Works great on the demo server. the 201 song instant mix might be excessive.  Maybe we can add a option in the settings section on how many songs you want to add to your instant mix. maybe add a pin pad and set_user_setting()

File: source/Main.brs (hunk 1 above)
Comment (1hitsong) [reply source/Main.brs:209]: :smile:

File: source/api/Items.brs (hunk 2 above)
Comment (1hitsong) [reply source/api/Items.brs:217]: The web client gathers a max of 200 songs for an Instant Mix. The methodology I use to get the proper ID will cause the 1st song on the album to always be the first song in the mix. I resolve this by requesting 201 and skipping the first song.

**Review by jimdogx — APPROVED** [review]

**Pull request merged**